### PR TITLE
Upgrade  to pattern library

### PR DESCRIPTION
**Pull request by boxadesign**
# Motivation and Context
Upgrade the response-ops repo to make proper use of the pattern library, and extract unnecessary code out.

# What has changed
* CSS removed that is also present in the pattern library
* A number of templates refactored to use pattern library markup
* A number of social code files, and their tests, removed to avoid fixing unit tests on unused code.

# How to test?
* Ensure the response-ops repo will run
* Ensure the response-ops repo units tests pass
* Run through user journeys to make sure that no UI regressions have occurred, and no journeys now fail.

# Links
https://trello.com/c/70JT2dJk/468-bring-in-pattern-library-to-response-ops-m

### CODE REVIEW

File: response_operations_ui/controllers/case_controller.py [hunk 1]
```diff
@@ -173,39 +143,3 @@ def generate_iac(case_id):
...
-        raise ApiError(response)
-
-    iac_count = len(response.json())
-
-    logger.debug("IAC count for case", case_id=case_id, url=url, iac_count=iac_count)
-
-    return iac_count
-
-
-def get_case_events_by_case_id(case_id):
```
Comment (shahi645) [inline response_operations_ui/controllers/case_controller.py]: This is actually needed as part of my code for the view response time

File: response_operations_ui/templates/change-response-status.html [hunk 2]
```diff
@@ -1,86 +1,71 @@
-{% extends "layouts/base.html" %}
+{% extends "layouts/spoke.html" %}
```
Comment (shahi645) [inline response_operations_ui/templates/change-response-status.html]: should this still not be base?

File: response_operations_ui/templates/confirm-enrolment-change.html [hunk 3]
```diff
@@ -50,8 +50,8 @@ <h1 class="jupiter">
         </dd>
       </dl>
 
-      <button type="submit" id="confirm-btn" class="btn btn--primary u-mt-s">
-        {% if change_flag == 'DISABLED' %}Disable{% else %}Enable{% endif %} enrolment
+      <button type="submit" id="confirm-btn" class="btn u-mb-s">
```
Comment (shahi645) [inline response_operations_ui/templates/confirm-enrolment-change.html]: 👍

File: response_operations_ui/controllers/case_controller.py (hunk 1 above)
Comment (owencjones) [reply response_operations_ui/controllers/case_controller.py]: Whereabouts is that code mate?

File: response_operations_ui/static/css/components/button.css [hunk 4]
```diff
@@ -1,4 +1,4 @@
-.btn{
+/* .btn{
```
Comment (NickGrantONS) [inline response_operations_ui/static/css/components/button.css]: did you mean to leave this commentted out?

File: response_operations_ui/static/css/override.css [hunk 5]
```diff
@@ -1,6 +1,6 @@
 /* Override styles - temporary for prototype */
 
-@import "/css/components/breadcrumb.css";
+/* @import "/css/components/breadcrumb.css";
```
Comment (NickGrantONS) [inline response_operations_ui/static/css/override.css]: again, did you mean to leave this commented out?

File: response_operations_ui/templates/messages.html [hunk 6]
```diff
@@ -28,13 +28,19 @@ <h1 class="jupiter" name="page-messages-title">Messages</h1>
...
+            <ul class="nav__list" aria-label="Section navigation menu">
+                <li class="nav__item pluto {% if not is_closed %}nav__item--current{% endif %}">
+                    <a class="nav__link" href="{{ url_for('messages_bp.view_selected_survey', selected_survey=selected_survey) }}" aria-current="{% if not is_closed %}location{% endif %}" role="menuitem">Open</a>
+                </li>
+                <li class="nav__item pluto {% if is_closed %}nav__item--current{% endif %}">
+                    <a class="nav__link" href="{{ url_for('messages_bp.view_selected_survey', selected_survey=selected_survey, is_closed='true') }}" aria-current="{% if is_closed %}location{% endif %}" role="menuitem">Closed</a>
+                </li>
+            </ul>
+        </div>
+    </nav>
```
Comment (NickGrantONS) [inline response_operations_ui/templates/messages.html]: indent out of alignment (pedantic sorry)

File: response_operations_ui/templates/partials/header.html [hunk 7]
```diff
@@ -1,33 +1,60 @@
...
-          </li>
-        </ul>
+<header class="header header--internal header--thin u-mb-no">
+  <div class="header__top" role="banner">
+    <div class="container container--wide">
+      <div class="grid grid--gutterless">
+        <div class="grid__col col-6@s">
+          <div class="logo">
+            {% set logo_print_cdn = cdn_url_prefix~"/img/ons-logo-black.svg" %}
+            <img src="{{ logo_print_cdn }}" alt="Office for National Statistics logo" class="logo__img header__logo--print">
```
Comment (NickGrantONS) [inline response_operations_ui/templates/partials/header.html]: alt text should be "Office for National Statistics" (no " logo") if we want to be consistent with the homepage

File: response_operations_ui/templates/partials/header.html [hunk 8]
```diff
@@ -1,33 +1,60 @@
...
+            <a href="/">
+              {% set logo_cdn = cdn_url_prefix~"/img/ons-logo-white.svg" %}
+              <img src="{{ logo_cdn }}" alt="Office for National Statistics logo" class="logo__img header__logo">
```
Comment (NickGrantONS) [inline response_operations_ui/templates/partials/header.html]: alt text should be "Office for National Statistics" (no " logo") if we want to be consistent with the homepage

File: response_operations_ui/templates/change-response-status.html [hunk 9]
```diff
@@ -1,86 +1,71 @@
...
+                        <label class="label label--inline venus " for="state-{{loop.index}}">{{ status }}</label>    
                     </div>
-                </div>
-                <div class="grid__col col-3@l">
-                </div>
+                    {% endfor %}
+                </fieldset>
             </div>
-        </div>
+            <button type="submit" id="response-status-change-confirm-button" class="btn btn--primary u-mt-s">Confirm</button>
```
Comment (NickGrantONS) [inline response_operations_ui/templates/change-response-status.html]: aren't you getting rid of all the references to `btn--primary`?

File: response_operations_ui/templates/layouts/base-head.html [hunk 10]
```diff
@@ -1,5 +1,10 @@
 <!DOCTYPE html>
-<html lang="en">
+{% set cdn_hash = "v1.5.3" %}
+{% set cdn_url_prefix = "https://cdn.ons.gov.uk/sdc/"~cdn_hash %}
+{% set css_responsive_cdn = cdn_url_prefix~"/css/responsive.css" %}
+{% set scripts_cdn = cdn_url_prefix~"/scripts/bundle.js" %}
```
Comment (NickGrantONS) [inline response_operations_ui/templates/layouts/base-head.html]: not used?

Comment (NickGrantONS) [reply response_operations_ui/templates/layouts/base-head.html]: ignore me... i see it in footer

**Comment by shahi645** [comment]
@NickGrantONS yes we got the go ahead from @davidmort

File: response_operations_ui/controllers/case_controller.py [hunk 11]
```diff
@@ -175,29 +145,10 @@ def generate_iac(case_id):
...
-
-    iac_count = len(response.json())
-
-    logger.debug("IAC count for case", case_id=case_id, url=url, iac_count=iac_count)
-
-    return iac_count
-
-
 def get_case_events_by_case_id(case_id):
     logger.debug('Retrieving cases', case_id=case_id)
```
Comment (shahi645) [inline response_operations_ui/controllers/case_controller.py]: Not tested, but i am adding tests in my PR

**Comment by shahi645** [comment]
Just need to fully regression test this, but good with changes so far. Breaking Acceptance Tests, will need some id's updating in them, but can take you through that

File: response_operations_ui/controllers/case_controller.py (hunk 11 above)
Comment (shahi645) [reply response_operations_ui/controllers/case_controller.py]: 🌚

**Comment by owencjones** [comment]
Closing for alternative PR

**Comment by shahi645** [comment]
Reopening as #257 has been merged

File: response_operations_ui/templates/collection-exercise.html [hunk 12]
```diff
@@ -0,0 +1,548 @@
+{% extends "layouts/base.html" %}
+{% block page_title %}{{survey.surveyRef}} {{survey.shortName}} {{ce.exerciseRef}} | Surveys | Survey Data Collection{% endblock %}
```
Comment (ryangrundy7) [inline response_operations_ui/templates/collection-exercise.html]: Is there a need for the `collection_exercises` directory in `templates`now, looks like you've merged them all into one file?
I wouldn't think of merging all of this into one file for this PR, doesn't look like it's be used anywhere?

File: response_operations_ui/templates/confirm-enrolment-change.html [hunk 13]
```diff
@@ -50,8 +50,8 @@ <h1 class="jupiter">
...
+        {% if change_flag == 'DISABLED' %}Disable{% else %}Enable{% endif %} enrolment?
```
Comment (gavinleeedwards) [inline response_operations_ui/templates/confirm-enrolment-change.html]: Tiny thing but there's a question mark that's now appeared on the button to disable an account - different to all other buttons.

Comment (owencjones) [reply response_operations_ui/templates/confirm-enrolment-change.html]: Addressing in next commit.

File: response_operations_ui/templates/collection-exercise.html (hunk 12 above)
Comment (owencjones) [reply response_operations_ui/templates/collection-exercise.html]: Addressed in next commit

Comment (owencjones) [reply response_operations_ui/templates/collection-exercise.html]: It seems to be used for exporting, so I've put it back.

**Pull request merged**